module/platform/OpenCR/HardwareIO/src/HardwareIO.cpp

```diff
@@ -0,0 +1,276 @@
+#include "HardwareIO.hpp"
+
+#include <fmt/format.h>
+
+#include "Convert.hpp"
+#include "dynamixel/v2/Dynamixel.hpp"
+
+#include "extension/Configuration.hpp"
+
+#include "message/actuation/ServoTarget.hpp"
+
+#include "utility/math/angle.hpp"
+#include "utility/math/comparison.hpp"
+#include "utility/support/yaml_expression.hpp"
+
+#define DEBUG_ENABLE_BUTTON_SPOOF 1
+
+namespace module::platform::OpenCR {
+
+    using extension::Configuration;
+    using message::actuation::ServoTarget;
+    using message::actuation::ServoTargets;
+    using message::platform::RawSensors;
+    using message::platform::StatusReturn;
+    using utility::support::Expression;
+
+    using message::platform::ButtonMiddleDown;
+
+    HardwareIO::HardwareIO(std::unique_ptr<NUClear::Environment> environment)
+        : Reactor(std::move(environment)), opencr(), nugus(), byte_wait(0), packet_wait(0), packet_queue() {
+
+        on<Configuration>("HardwareIO.yaml").then([this](const Configuration& config) {
+            this->log_level = config["log_level"].as<NUClear::LogLevel>();
+
+            // Make sure OpenCR is operating at the correct baud rate (based on config params)
+            if (opencr.connected()) {
+                opencr.close();
+            }
+
+            opencr.open(config["opencr"]["device"], config["opencr"]["baud"]);
+            byte_wait   = config["opencr"]["byte_wait"];
+            packet_wait = config["opencr"]["packet_wait"];
+
+            // Initialise packet_queue map
+            // OpenCR
+            packet_queue[uint8_t(NUgus::ID::OPENCR)] = std::vector<PacketTypes>();
+            // Servos - NOT zero indexed. First servo is ID 1.
+            // Required because response packets have Dynamixel ID
```

Comment:
They aren't zero indexed because the packet queue tracks packets based on the dynamixel ID of the device it's communicating with i.e. packet_queue[1] is the queue for ID 1 etc.

Reply:
I've updated the comment to hopefully make it make more sense